### ISSUE TITLE
Long dataframe

### ISSUE THREAD

**Issue by xianwenchen** [frederic-santos/ob-julia#2]
Thank you very much for picking up the ob-julia.

It is great to see that it is working!

I have a small issue with a long dataframe.
Here is the reproducible code:

```
#+BEGIN_SRC ess-julia :session julia :exports both
dtf = DataFrame(
    [8.04   9.14  7.46   6.58;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     6.95   8.14  6.77   5.76;
     7.58   8.74  12.74  7.71;
     8.81   8.77  7.11   8.84;
     8.81   8.77  7.11   8.84;
     8.33   9.26  7.81   8.47])
showall( dtf )
#+END_SRC
```

org-mode does not produce any output on my machine.

org-mode does not produce any output if I remove the *showall( dtf )* line.

Could you check if it is working on yours?

Merry Christmas!

**Comment by frederic-santos** [comment]
Hi,

Thanks for this feedback! I tested this code block: for me it works fine for an external ("non-session") eval with both `:results value` and `:results output`; it also works fine for a session eval *only with* `:results value`. But a session eval with `:results output` froze the inferior ESS session, so that no output was produced (with or without `showall()`, this did not change anything for me).

I'll give a closer look on that problem soon!

**Comment by xianwenchen** [comment]
Thanks! I think being able to use session is useful, when different source block use the same set of variables. It will be great if you could look at it. I appreciate your inputs!

**Comment by frederic-santos** [comment]
A first hypothesis here. Actually, in `ob-ess-julia`, each instruction is rewritten in one single line (to overcome some problems due to ESS). Thus, your instruction is first translated as:
```julia
dtf = DataFrame(     [8.04   9.14  7.46   6.58;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      6.95   8.14  6.77   5.76;      7.58   8.74  12.74  7.71;      8.81   8.77  7.11   8.84;      8.81   8.77  7.11   8.84;      8.33   9.26  7.81   8.47])
```

Try to execute this instruction in an ESS session, in a classical `.jl` buffer. It seems that this instruction (because it's too "big"?) cannot be properly executed: at least on my computer, ESS freezes.
Conversely, your original instruction (with all the newlines) can be executed without any problem by ESS. Thus, it won't be easy to solve that bug on my side... (If this does not work on your computer either, opening an issue on the ESS repo would be useful.)

This was a true problem for me: `ob-ess-julia` "inherits" all the glitches and bugs from ESS, which has not a perfect Julia support (the ESS manual is very clear about that). Many very dirty parts in my elisp code were only workarounds to avoid several ESS bugs. This "pre-processing", that rewrites each Julia instruction on one single line, was necessary to avoid some of them... but now it triggers another bug, which seems to be an ESS bug. Looks complicated. :smile: 

I let this issue open, but I can't say that I'm sure to find a good solution for it...

(Just an additional remark: for the future, I would clearly bet on [ob-julia-vterm](https://github.com/shg/ob-julia-vterm.el), which is not based on ESS, and relies on the very solid `vterm`. It has to deal with much fewer intrinsic limitations; so I think that it could become the best available solution for Julia in org-mode.)

**Comment by xianwenchen** [comment]
Thank you very much!

I spent last night and early morning to install ob-julia-vterm. It indeed worked!

I don't know why ob-ess-julia was having the issue. However, I do not think that it was because your computer was slow. The data frame itself would require a very small amount of memory.

Cheers!

**Comment by frederic-santos** [comment]
Hi @xianwenchen ,
This is solved with commit 429097c, and v1.0.2. The code block is now successfully evaluated. However:
* if you want to use `showall(dtf)` at the end of the code block, you'll have to say `:results output` in the header
* if you want to use `:results value` in the header, replace `showall(dtf)` by simply `dtf`.

Thanks for the report!

**Comment by xianwenchen** [comment]
Thank you!
